src/components/video-player/useVideoContainerHook.ts

```diff
@@ -10,14 +10,17 @@ import useIntersection from 'react-use/lib/useIntersection';
 import useUnmount from 'react-use/lib/useUnmount';
 
 import { useVideo } from '../../hooks';
+import { ExtendedTimeUpdateEvent } from '../../types';
 import { OVERLAY_HIDE_DELAY, PROGRESS_INTERVAL } from '../../utils/constants';
 import { getElementOffset } from '../../utils/html-elements';
 import { ContainerSizePosition } from '../draggable-popover/DraggablePopover';
 
-interface UseVideoContainerHookProps {
+export interface UseVideoContainerHookProps {
 	videoUrl: string;
+	/** Boolean that represents if the play is enabled. */
 	hasPlayEnabled?: boolean;
 	onPlay?: VoidFunction;
+	onTimeUpdate?: (event: ExtendedTimeUpdateEvent) => void;
```

Comment:
(Probably more to @bragaru-i ) I'm unclear when to use `videoContext.addListener` vs `onXXX`. The announcement post indicated that `addListener` is the right way but now we also expose functions like `onPlay`. What's the right way here?

Reply:
Adding `onPlay`  to `VideoPlayerProps` for not  moving all the logic outside + `onPlay` function shouldn't change so often to create a rerenders.

`onTimeUpdate` will be just be triggered frequently in VideoContext and create rerenders ....
So I would like to suggest to use it in this way:
```ts
const MyComponent = () => {
	const videoContextRef = React.useRef<VideoContext>();
	const videoContextApi = videoContextRef.current?.api;

	const setVideoContext = React.useCallback(
		(context: VideoContext) => {
			videoContextRef.current = context;
		},
		[videoContextRef],
	);

	React.useEffect(() => {
		const someFn = e => {
			console.log(e);
			
		};
		videoContextApi?.addEventListener?.('timeupdate', someFn);

		return () => videoContextApi?.removeEventListener?.('timeupdate', someFn);
	}, [videoContextApi]);

	return <VideoPlayer videoUrl="url" onContext={setVideoContext} />;
};
```
In this way, we would have minimal implications in the video player logic from outside + we will get MyComponent rerender only when `timeupdate` runs

Reply:
I am now more concerned too get rid of the `onPlay`, and move its logic  in the component that will just listen `play` event

Reply:
@bragaru-i  The PR was created because `useEventListener` didn't work. What is suggested here didn't work too